src/shared/ui/survey/AudioPlayer.tsx

```diff
@@ -0,0 +1,80 @@
+import { FC, useState, useCallback } from 'react';
+import { TouchableOpacity, StyleSheet } from 'react-native';
+
+import { useFocusEffect } from '@react-navigation/native';
+import AudioRecorderPlayer from 'react-native-audio-recorder-player';
+
+import { PlayIcon, PauseIcon, XStack, Text } from '@shared/ui';
+
+type Props = {
+  uri: string;
+  title?: string;
+};
+
+const audioRecorderPlayer = new AudioRecorderPlayer();
+
+const AudioPlayer: FC<Props> = ({ uri, title }) => {
+  const [playing, setPlaying] = useState(false);
+
+  useFocusEffect(
+    useCallback(() => {
+      return () => {
+        setPlaying(false);
+        audioRecorderPlayer.stopPlayer();
+        audioRecorderPlayer.removePlayBackListener();
+      };
+    }, []),
+  );
+
+  const togglePlay = async () => {
+    if (playing) {
+      await pause();
+    } else if (!playing) {
+      await play();
+    }
+  };
+
+  const play = async () => {
+    setPlaying(true);
+
+    await audioRecorderPlayer.startPlayer(uri);
+    audioRecorderPlayer.addPlayBackListener(data => {
+      if (data.currentPosition === data.duration) {
+        setPlaying(false);
```

Comment:
We should probably call the `stop` function here also

Reply:
@BamMironov I think when this if block is entered the player stops by itself, it doesn't loop.

Reply:
Ok